modules/policy/src/main/java/org/eclipse/xpanse/modules/policy/policyman/PolicyManager.java

```diff
@@ -0,0 +1,309 @@
+import org.eclipse.xpanse.modules.security.IdentityProviderManager;
+import org.springframework.beans.BeanUtils;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.security.access.AccessDeniedException;
+import org.springframework.stereotype.Component;
+import org.springframework.util.CollectionUtils;
+import org.springframework.web.client.RestClientException;
+
+/**
+ * The service of policy-man.
+ */
+@Slf4j
+@Component
+public class PolicyManager {
+
+
+    @Resource
+    private DatabasePolicyStorage policyStorage;
+
+    @Resource
+    private AdminApi adminApi;
+
+    @Resource
+    private PoliciesValidateApi policiesValidateApi;
+
+    @Resource
+    private PoliciesEvaluationApi policiesEvaluationApi;
+
+    @Resource
+    private IdentityProviderManager identityProviderManager;
+
+    @Value("${policy.man.endpoint:http://localhost:8090}")
+    private String policyManBaseUrl;
+
+
+    /**
+     * Get system status of the policyMan.
+     *
+     * @return Returns BackendSystemStatus.
+     */
+    public BackendSystemStatus getPolicyManStatus() {
+        BackendSystemStatus policyManStatus = new BackendSystemStatus();
+        policyManStatus.setBackendSystemType(BackendSystemType.POLICY_MAN);
+        policyManStatus.setName(BackendSystemType.POLICY_MAN.toValue());
+        policyManStatus.setEndpoint(policyManBaseUrl);
+        try {
+            org.eclipse.xpanse.modules.policy.policyman.generated.model.SystemStatus
+                    policyManSystemStatus = adminApi.healthGet();
+            org.eclipse.xpanse.modules.policy.policyman.generated.model.HealthStatus
+                    healthStatus = policyManSystemStatus.getHealthStatus();
+            policyManStatus.setHealthStatus(HealthStatus.valueOf(healthStatus.getValue()));
+        } catch (RestClientException e) {
+            log.error("Get status of policy-man error:{}", e.getMessage());
+            policyManStatus.setHealthStatus(HealthStatus.NOK);
+            policyManStatus.setDetails(e.getMessage());
+        }
+
+        return policyManStatus;
+    }
+
+
+    /**
+     * List policies by the query model.
+     *
+     * @param queryModel The query model.
+     * @return Returns all policies matched the query model.
+     */
+    public List<PolicyVo> listPolicies(PolicyQueryRequest queryModel) {
+        queryModel.setMyUserId(getCurrentLoginUserId());
+        List<PolicyEntity> policyEntities = policyStorage.listPolicies(queryModel);
+        return policyEntities.stream().sorted(Comparator.comparing(PolicyEntity::getCsp))
+                .map(this::conventToPolicyVo).toList();
+    }
+
+    /**
+     * Create new policy.
+     *
+     * @param createRequest create policy request.
+     * @return Returns created policy view object.
+     */
+    public PolicyVo addPolicy(PolicyCreateRequest createRequest) {
+        checkPolicyWhetherCreated(createRequest.getCsp(), createRequest.getPolicy());
```

Comment:
Does it allow multiple policies for every CSP?

Reply:
It allows the same user to create multiple policies in the every CSP, but the policies created by the same user for the same CSP could not be duplicate.

Reply:
My expectation was that.. for every user, we must allow multiple policies per CSP.. Do you think we can achieve this easily? If not, let us discuss together with Yichao what he thinks.

Reply:
Now the APIs allow each user to create multiple policies per CSP. No need to make any changes again.